app/coffee/Router.coffee

```diff
@@ -80,13 +80,13 @@ module.exports = Router =
 						Router._handleError null, err, client, "leaveProject"
 
 
-			client.on "joinDoc", (doc_id, fromVersion, callback) ->
+			client.on "joinDoc", (doc_id, options, fromVersion, callback) ->
```

Comment:
We need to make this optional, since old clients will still be passing args with the old argument signature.

Reply:
Right of course. I'll fix that

Reply:
Thanks. I think `options` should be after `fromVersion`, since we may remove it after this is migrated. Also the end 'feels' like the right place for a flexible bundle of options (essentially named optional arguments).